godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Quat.kt

```diff
@@ -0,0 +1,379 @@
+@file:Suppress("unused", "MemberVisibilityCanBePrivate")
+
+package godot.core
+
+import godot.gdnative.godot_quat
+import godot.gdnative.godot_quat_layout
+import godot.gdnative.godot_vector3_layout
+import kotlinx.cinterop.*
+import kotlin.math.*
+
+class Quat(var x: RealT, var y: RealT, var z: RealT, var w: RealT) : CoreType {
+    //CONSTANTS
+    companion object {
+        val IDENTITY = Quat(0.0, 0.0, 0.0, 1.0)
+    }
+
+
+    //CONSTRUCTOR
+    constructor() :
+        this(0.0, 0.0, 0.0, 1.0)
+
+    constructor(x: Number, y: Number, z: Number, w: Number = 1.0) :
+        this(x.toRealT(), y.toRealT(), z.toRealT(), w.toRealT())
+
+    constructor(from: Basis): this() {
+        val trace = from[0][0] + from[1][1] + from[2][2]
+        val temp: DoubleArray
+
+        if (trace > 0.0) {
+            var s = sqrt(trace + 1.0)
+            val temp3 = s * 0.5
+            s = 0.5 / s
+            temp = doubleArrayOf(
+                ((from[2][1] - from[1][2]) * s),
+                ((from[0][2] - from[2][0]) * s),
+                ((from[1][0] - from[0][1]) * s),
+                temp3
+            )
+        } else {
+            temp = doubleArrayOf(0.0, 0.0, 0.0, 0.0)
+            val i = if (from[0][0] < from[1][1]) {
+                if (from[1][1] < from[2][2]) 2 else 1
+            } else {
+                if (from[0][0] < from[2][2]) 2 else 0
+            }
+            val j = (i + 1) % 3
+            val k = (i + 2) % 3
+
+            var s = sqrt(from[i][i] - from[j][j] - from[k][k] + 1.0)
+            temp[i] = s * 0.5
+            s = 0.5 / s
+            temp[3] = (from[k][j] - from[j][k]) * s
+            temp[j] = (from[j][i] + from[i][j]) * s
+            temp[k] = (from[k][i] + from[i][k]) * s
+        }
+        set(temp[0], temp[1], temp[2], temp[3])
+    }
+
+    constructor(axis: Vector3, angle: RealT): this() {
+        val d: RealT = axis.length()
+        if (d == 0.0) set(0.0, 0.0, 0.0, 0.0)
+        else {
```

Comment:
I know you are not a fan of {} but i find this hard to read but do as you please

Reply:
It's leftover from the previous bindings. I already removed a ton of them. I just forgot some of them. 
Don't worry I don't like it either.

Reply:
:+1: Thanks a lot!